subprojects/configuration-cache/src/integTest/groovy/org/gradle/configurationcache/ConfigurationCacheIntegrationTest.groovy

```diff
@@ -25,22 +25,74 @@ import spock.lang.Issue
 
 class ConfigurationCacheIntegrationTest extends AbstractConfigurationCacheIntegrationTest {
 
-    def "configuration cache for help on empty project"() {
+    def "configuration cache for Help plugin task '#task' on empty project"() {
         given:
         settingsFile.createFile()
-        configurationCacheRun "help"
+        configurationCacheRun(task, *options)
         def firstRunOutput = removeVfsLogOutput(result.normalizedOutput)
-            .replaceAll(/Calculating task graph as no configuration cache is available for tasks: help\n/, '')
+            .replaceAll(/Calculating task graph as no configuration cache is available for tasks: ${task}.*\n/, '')
             .replaceAll(/Configuration cache entry stored.\n/, '')
 
         when:
-        configurationCacheRun "help"
+        configurationCacheRun(task, *options)
         def secondRunOutput = removeVfsLogOutput(result.normalizedOutput)
             .replaceAll(/Reusing configuration cache.\n/, '')
             .replaceAll(/Configuration cache entry reused.\n/, '')
 
         then:
         firstRunOutput == secondRunOutput
+
+        where:
+        task            | options
+        "help"          | []
+        "properties"    | []
+        "dependencies"  | []
+        "help"          | ["--task", "help"]
+    }
+
+    def "can store task selection success/failure for :help --task"() {
+        def configurationCache = newConfigurationCacheFixture()
+        buildFile.text = """
+        task aTask
+"""
```

Comment:
💄 Indentation is off. See how we do on other tests.

Reply:
Indeed, I was following the example of `HelpTaskIntegrationTest`, which I guess does that to avoid the additional tabs/spaces in the last line. Fixing it.